src/Microsoft.AspNetCore.Sockets.Client/WebSocketsTransport.cs

```
@@ -0,0 +1,139 @@
+                    receiveResult = await _webSocket.ReceiveAsync(buffer, cancellationToken);
+                    incomingMessage.Add(buffer.Array);
+                    totalBytes += receiveResult.Count;
+                } while (!receiveResult.EndOfMessage);
+
+                Message message;
+                if (incomingMessage.Count > 1)
+                {
+                    var messageBuffer = new byte[totalBytes];
+                    var offset = 0;
+                    for (var i = 0 ; i < incomingMessage.Count; i++)
+                    {
+                        Buffer.BlockCopy(incomingMessage[i], 0, messageBuffer, offset, incomingMessage[i].Length);
+                        offset += incomingMessage[i].Length;
+                    }
+                    
+                    message = new Message(ReadableBuffer.Create(messageBuffer).Preserve(), receiveResult.MessageType == 0 ? Format.Text : Format.Binary, receiveResult.EndOfMessage);
+                }
+                else
+                {
+                    message = new Message(ReadableBuffer.Create(incomingMessage[0]).Preserve(), receiveResult.MessageType == 0 ? Format.Text : Format.Binary, receiveResult.EndOfMessage);
+                }
+
+                while (await _application.Output.WaitToWriteAsync(cancellationToken))
+                {
+                    if (_application.Output.TryWrite(message))
+                    {
+                        incomingMessage.Clear();
+                        break;
+                    }
+                }
+            }
+        }
+
+        private async Task SendMessages(Uri sendUrl, CancellationToken cancellationToken)
+        {
+            while (await _application.Input.WaitToReadAsync(cancellationToken))
+            {
+                Message message;
+                while (!cancellationToken.IsCancellationRequested && _application.Input.TryRead(out message))
```

I don't like this extra condition, let the thing throw an operationcancelledexception